### PR TITLE
Editor: Migrate deprecated React.PropTypes to npm prop-types module.

### PR DESCRIPTION
**Pull request by iandunn**
This was originally part of #16819, but that PR was getting too big, so I'm breaking it into multiple ones.

### CODE REVIEW

File: client/post-editor/editor-featured-image/index.jsx [hunk 1]
```diff
@@ -26,12 +27,14 @@ import { recordTracksEvent } from 'state/analytics/actions';
...
-		selecting: React.PropTypes.bool,
-		onImageSelected: React.PropTypes.func,
-		featuredImage: React.PropTypes.object,
+		maxWidth: PropTypes.number,
+		site: PropTypes.object,
+		post: PropTypes.object,
+		selecting: PropTypes.bool,
+		onImageSelected: PropTypes.func,
+		featuredImage: PropTypes.object,
+		hasDropZone: PropTypes.bool,
```
Comment (gziolo) [inline client/post-editor/editor-featured-image/index.jsx]: I don't see `hasDropZone` and `isDropZoneVisible` inside this file. Is this coming from other PR? :)

File: client/components/tinymce/index.jsx [hunk 2]
```diff
@@ -161,29 +162,29 @@ module.exports = React.createClass( {
...
+		onHide: PropTypes.func,
+		onInit: PropTypes.func,
+		onRedo: PropTypes.func,
+		onRemove: PropTypes.func,
+		onReset: PropTypes.func,
+		onShow: PropTypes.func,
+		onSubmit: PropTypes.func,
+		onUndo: PropTypes.func,
+		onSetContent: PropTypes.func,
+		tabIndex: PropTypes.number,
```
Comment (nosolosw) [inline client/components/tinymce/index.jsx]: The EVENTS array contains the `init` and `mouseUp` events, should we add them here as well?

I tend to prefer grouping all the event handlers together at the end, being the first props all non-functions. That way I can have a quick feeling for the kind of events the component supports by just looking at the eventHandlers properties.

File: client/components/tinymce/index.jsx [hunk 3]
```diff
@@ -161,29 +162,29 @@ module.exports = React.createClass( {
...
+		isNew: PropTypes.bool,
+		onTextEditorChange: PropTypes.func,
+		onKeyUp: PropTypes.func
```
Comment (nosolosw) [inline client/components/tinymce/index.jsx]: In the next days, we'll merge a PR with an updated ESLint version and rules. The [comma-dangle](https://eslint.org/docs/rules/comma-dangle) will be enforced from then on, so let's add a comma here.

File: client/post-editor/editor-featured-image/index.jsx [hunk 4]
```diff
@@ -26,12 +27,14 @@ import { recordTracksEvent } from 'state/analytics/actions';
...
+		isDropZoneVisible: PropTypes.bool,
```
Comment (nosolosw) [inline client/post-editor/editor-featured-image/index.jsx]: We are missing here the `recordTrackEvents` and `translate` props (the later is being added by the localize Higher Order Component). Also, are we using `hasDropZone` and `isDropZoneVisible`?

File: client/post-editor/editor-html-toolbar/index.jsx [hunk 5]
```diff
@@ -1,7 +1,8 @@
 /**
  * External dependencies
  */
-import React, { Component, PropTypes } from 'react';
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
```
Comment (nosolosw) [inline client/post-editor/editor-html-toolbar/index.jsx]: I think we are missing the `site` prop in PropTypes for this component.

File: client/post-editor/editor-featured-image/index.jsx (hunk 1 above)
Comment (iandunn) [reply client/post-editor/editor-featured-image/index.jsx]: Ah, you're right, I'll remove those, good catch :)

File: client/components/tinymce/index.jsx (hunk 2 above)
Comment (iandunn) [reply client/components/tinymce/index.jsx]: It looks like `onInit` is already there, but I'll add `onMouseOver` and re-organize the handlers at the bottom. Thanks :)

Comment (nosolosw) [reply client/components/tinymce/index.jsx]: Oh, sorry for the confusion, I meant, `onInput` not `onInit`!

Comment (iandunn) [reply client/components/tinymce/index.jsx]: Ah, thanks, I'll add that as well.

**Pull request merged**